app/main/views/g_cloud.py

```diff
@@ -605,8 +605,38 @@ def which_service_won_contract(framework_framework, project_id):
     '/<string:framework_framework>/projects/<int:project_id>/why-didnt-you-award-contract',
     methods=['GET']
 )
-def why_didnt_you_award_contract(framework_framework, project_id):
-    abort(404)
+def why_did_you_not_award_the_contract(framework_framework, project_id):
+    all_frameworks = data_api_client.find_frameworks().get('frameworks')
+    framework = framework_helpers.get_latest_live_framework(
+        all_frameworks, framework_framework)
+    
+    project = data_api_client.get_direct_award_project(project_id=project_id)['project']
+
+    form = WhyDidYouNotAwardForm()
+
+    if request.method == "POST" and form.validate_on_submit():
+        flash('You’ve updated ‘' + project.name + '’')
+        return redirect(url_for('.view_project', framework_framework=framework_framework, project_id=project['id']))
+
+    form.why_did_you_not_award_the_contract.options = [
+        {
```

Comment:
I'm a bit confused here why we need to update `options`, presumably overriding the `choices` already set? in any even, it feels like these should be in the form constructor, shouldn't they?

Reply:
you're right i'll put it in as a form constructor. The reason why I needed it because the template needs the description value passed in

Reply:
@DilwoarH have you seen `DMRadioField`? It's a class added it to `direct_award_forms.py`, it creates the `options` in the correct format for you

Reply:
I put it inside the __init__ function now - can't use DMRadioField as it does not allow descriptions to be passed in and it uses the choices which only has label and value in it's array

Reply:
Ah, understood. I should move `DMRadioField` to `dmutils` anyway so I'll look at adding that functionality.